infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/local/LocalCheInfrastructureProvisioner.java

```diff
@@ -82,56 +88,93 @@ public LocalCheInfrastructureProvisioner(InstallerConfigApplier installerConfigA
     }
 
     @Override
-    public void provision(Environment envConfig,
+    public void provision(EnvironmentImpl envConfig,
                           DockerEnvironment internalEnv,
                           RuntimeIdentity identity)
             throws InfrastructureException {
         String devMachineName = getDevMachineName(envConfig);
         if (devMachineName == null) {
             throw new InfrastructureException("ws-machine is not found on installers applying");
         }
+        excludeDirsFromSnapshot(internalEnv);
+        addProjects(internalEnv, identity.getWorkspaceId(), devMachineName);
```

Comment:
These methods do what their names say. `addProjects` adds everything needed for projects folder. `addInstallers` adds everything needed for installers, which may include volumes, env variables, etc. 
This was the motivation for the creation of these methods.